x/sync/manager.go

```diff
@@ -604,20 +604,28 @@ func (m *Manager) completeWorkItem(ctx context.Context, work *workItem, largestH
 		}
 	}
 
-	// completed the range [work.start, lastKey], log and record in the completed work heap
-	m.config.Log.Info("completed range",
-		zap.Binary("start", work.start),
-		zap.Binary("end", largestHandledKey),
-	)
-	if m.getTargetRoot() == rootID {
+	// Process [work] while holding [syncTargetLock] to ensure that object
+	// is added to the right queue, even if a target update is triggered
+	m.syncTargetLock.RLock()
```

Comment:
Is there a reason we can't `defer` the unlock of this?

Reply:
Was trying to unlock before we emitted log

Reply:
feels premature tbh (esp for a RLock)

Reply:
will change